### PR TITLE
added 'An Anarchist FAQ' (AFAQ) package.

### PR DESCRIPTION
**Pull request by oida**
###### Motivation for this change

Anarchy!

###### Things done

- [x ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x ] Tested execution of all binary files (usually in `./result/bin/`)
- [x ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -1216,7 +1216,7 @@ with pkgs;
   mpdris2 = callPackage ../tools/audio/mpdris2 { };
 
   nfdump = callPackage ../tools/networking/nfdump { };
-
```
Comment (catern) [inline pkgs/top-level/all-packages.nix]: Unneeded whitespace damage?

File: pkgs/misc/anarchism/default.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    sha256 = "b081885152628c67c0f6d9101be17095d84684bc9dd2c7d4e50f5744614883ef";
+  };
+
+  installPhase = ''
+    mkdir -p $out/share/doc/anarchism/html
+    mkdir $out/share/doc/anarchism/markdown
+    mkdir -p $out/share/applications/
+    cp -r html $out/share/doc/anarchism/
+    cp -r markdown $out/share/doc/anarchism/
+    echo \[Desktop Entry\] > $out/share/applications/anarchism.desktop
```
Comment (yegortimoshenko) [inline pkgs/misc/anarchism/default.nix]: Use `makeDesktopItem`. Here are examples: https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=makeDesktopItem&type=

File: pkgs/misc/anarchism/default.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+stdenv.mkDerivation rec {
+  name = "anarchism-${version}";
+  version = "15.0";
+  src = fetchurl {
+    url = https://0xacab.org/ju/afaq/repository/master/archive.tar.gz;
+    sha256 = "b081885152628c67c0f6d9101be17095d84684bc9dd2c7d4e50f5744614883ef";
+  };
+
+  installPhase = ''
+    mkdir -p $out/share/doc/anarchism/html
```
Comment (yegortimoshenko) [inline pkgs/misc/anarchism/default.nix]: Is this really documentation, though?

File: pkgs/misc/anarchism/default.nix [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    mkdir $out/share/doc/anarchism/markdown
+    mkdir -p $out/share/applications/
+    cp -r html $out/share/doc/anarchism/
+    cp -r markdown $out/share/doc/anarchism/
```
Comment (yegortimoshenko) [inline pkgs/misc/anarchism/default.nix]: Better just `mv`, that would make it clearer:

```
mkdir -p $out/share/doc/anarchism
mv html markdown $out/share/doc/anarchism
```

File: pkgs/misc/anarchism/default.nix [hunk 5]
```diff
@@ -0,0 +1,42 @@
+{ stdenv, fetchurl }:
+
+stdenv.mkDerivation rec {
+  name = "anarchism-${version}";
+  version = "15.0";
+  src = fetchurl {
+    url = https://0xacab.org/ju/afaq/repository/master/archive.tar.gz;
```
Comment (yegortimoshenko) [inline pkgs/misc/anarchism/default.nix]: This will break very soon. You should provide a link to a stable archive, i.e. one that will never change. Or use `fetchgit`.

**Review by lukateras — CHANGES_REQUESTED** [review]
Heh :-) Seriously, though, should we package political books in Nixpkgs? I'd rather not do that.

File: pkgs/misc/anarchism/default.nix [hunk 6]
```diff
@@ -0,0 +1,42 @@
+{ stdenv, fetchurl }:
+
+stdenv.mkDerivation rec {
+  name = "anarchism-${version}";
```
Comment (yegortimoshenko) [inline pkgs/misc/anarchism/default.nix]: `anarchism` is not a proper name for this work. It should be `afaq`, or `an-anarchist-faq`.

**Comment by yegortimoshenko** [comment]
Sorry, I don't think this is a good idea.

File: pkgs/misc/anarchism/default.nix [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+  meta = with stdenv.lib; {
+    description = "Exhaustive exploration of Anarchist theory and practice.";
+    longDescription = ''
+      The Anarchist FAQ is an excellent source of information regarding
+      Anarchist (libertarian socialist) theory and practice. It covers all major
+      topics, from the basics of Anarchism to very specific discussions of
+      politics, social organization, and economics.
+    '';
+    homepage = http://www.anarchistfaq.org;
+    license = stdenv.lib.licenses.gpl2;
```
Comment (yegortimoshenko) [inline pkgs/misc/anarchism/default.nix]: I don't see any notice on AFAQ site that it is under GPLv2.

File: pkgs/misc/anarchism/default.nix (hunk 5 above)
Comment (oida) [reply pkgs/misc/anarchism/default.nix]: done!

File: pkgs/misc/anarchism/default.nix (hunk 3 above)
Comment (oida) [reply pkgs/misc/anarchism/default.nix]: yes

File: pkgs/misc/anarchism/default.nix (hunk 4 above)
Comment (oida) [reply pkgs/misc/anarchism/default.nix]: done.

File: pkgs/misc/anarchism/default.nix (hunk 2 above)
Comment (oida) [reply pkgs/misc/anarchism/default.nix]: Did it, had to change the name tho as that fails with spaces, don't like it. :(

File: pkgs/misc/anarchism/default.nix (hunk 6 above)
Comment (oida) [reply pkgs/misc/anarchism/default.nix]: It has been for many years in debian world, on archlinux and whatnot. It's a well known package and often refered to by it's pkg name and I'd like to keep it that way.

File: pkgs/misc/anarchism/default.nix (hunk 7 above)
Comment (oida) [reply pkgs/misc/anarchism/default.nix]: Both can be applied: "Permission is granted to copy, distribute and/or modify this document under the terms of the GNU Free Documentation License, Version 1.1 or any later version published by the Free Software Foundation, and/or the terms of the GNU General Public License, Version 2.0 or any later version published by the Free Software Foundation." http://www.anarchism.pageabode.com/afaq/intro.html

File: pkgs/misc/anarchism/default.nix (hunk 6 above)
Comment (yegortimoshenko) [reply pkgs/misc/anarchism/default.nix]: Context: https://lists.debian.org/debian-devel/1999/03/msg01628.html

Including this package in Debian has arguably been a mistake that I wish will not be repeated here. Software distributions are not the proper place for political debate.

**Comment by fpletz** [comment]
@yegortimoshenko Closing this PR and limiting the conversion to collaborators without any discussion only based on your opinion is ridiculous. You're not not speaking for the whole project and please do not try to do so.

**Comment by fpletz** [comment]
As I somehow cannot unlock this PR: @oida please open a new PR if you still intend to include the package.

**Comment by gebner** [comment]
Interestingly, I was able to unlock it.

I'm not sure if this package fits into scope of nixpkgs, though.  We don't package offline builds of wikipedia either.

**Comment by vcunat** [comment]
IMO packaging political books serves little purpose except as "publicity" for the programme (books tend to be easy to "just download").  I'd prefer to avoid spending energy on politics _around nixos.org_, as I can't see any sufficient relation.

**Comment by yegortimoshenko** [comment]
> Closing this PR and limiting the conversion to collaborators without any discussion only based on your opinion is ridiculous. 

I don't think so. I've closed this because I feel confident in my decision and I don't believe this to be worth time of other members. Nod from @grahamc seems to be enough.

Limiting conversation to collaborators is exactly that, i.e. you can still respond. I've locked it because I don't want this to evolve into a political debate (there are no practical questions left unasked), and I ask you to lock it again.

Political books and discussions are divisive by their nature, as in, they tend to impose enemies on the readers and divide world into "us" and "them". I think everyone would agree, no matter what their political views are, that software distribution is not a proper place for that, and if anything, it 
would result in a less welcoming and more divisive community for all.

Also, tarballs are better way to store data than in Nix store, because tarballs don't have a build program attached to them, they are purely declarative.

**Comment by yegortimoshenko** [comment]
Also, I have never implied that I speak for the whole project. Not sure why you think otherwise.

**Comment by catern** [comment]
Using Nix to package/distribute books and non-technical documents could be interesting. If someone wants to experiment with it, they should do it in a separate repository of "Nixbooks" (possibly as a Nixpkgs overlay), and if it proves itself as useful, it could be merged into Nixpkgs. I think @oida should do that if they want to continue with this derivation.

The fact that this is a political work is a distraction; even if it wasn't political, it wouldn't make sense to merge, because it's not clear that it makes sense to distribute books through Nix at all. And if it turns out to make sense to distribute books through Nix, then the political content of the books would be irrelevant.

**Comment by yegortimoshenko** [comment]
> And if it turns out to make sense to distribute books through Nix, then the political content of the books would be irrelevant.

That would only be true if genuinely no particular criteria were applied to accepting books other than willingness of someone to maintain them. It seems highly unrealistic to me, and distributed systems like Web or IPFS solve this much better by not relying on any single authority for accepting data.

Distributed systems also work around censorship issue, where some books that are acceptable in some jurisdictions but are banned in others.

In other words, Nixbooks would never work for political material.

Software for the most part doesn't pose ethical/moral issues in itself and is only concerned about practicality. Games, movies and fiction all also have a practical purpose, that of enjoyment. Political material is inherently different because it incentivizes action in the real world, and divides people into friends and enemies, the latter being more pronounced in radical books like one in this pull request.
